### PR TITLE
[Jormungandr] Manage realtime proxies from database

### PR DESCRIPTION
**Pull request by azime**
Jira: [NAVP-1831](https://jira.kisio.org/browse/NAVP-1831)

### CODE REVIEW

File: source/jormungandr/jormungandr/instance.py [hunk 1]
```diff
@@ -213,7 +214,15 @@ def get_external_service_providers_from_db(self):
         """
         :return: a callable query of external services associated to the current instance in db
         """
-        return self._get_models().external_services
+        result = self._get_models().external_services
+        return [res for res in result if res.navitia_service != 'realtime_proxies']
```
Comment (kadhikari) [inline source/jormungandr/jormungandr/instance.py]: It seems fine now but when we add a new service, we will have to modify here and below. What I propose:
return [res for res in result if res.navitia_service in ['free_floatings', 'vehicle_occupancies']]

**Review by kadhikari — COMMENTED** [review]
It seems fine to me. Could you test in local the followings:

- Only config in file
- One config in file and another in db but different id
- Same config in the file and also in db with one parameter value modified and verify that the new modified value in db is used.
- Config only in db

File: source/jormungandr/jormungandr/instance.py (hunk 1 above)
Comment (azime) [reply source/jormungandr/jormungandr/instance.py]: OK

**Review by kadhikari — APPROVED** [review]

File: source/jormungandr/jormungandr/realtime_schedule/realtime_proxy_manager.py [hunk 2]
```diff
@@ -31,57 +31,136 @@
...
+        try:
+            object_id_tag = rt_proxy.args.get('object_id_tag', rt_proxy.id)
+            if 'object_id_tag' in rt_proxy.args:
+                del rt_proxy.args["object_id_tag"]
+            self._realtime_proxies[rt_proxy.id] = self._init_class(
+                rt_proxy.id, rt_proxy.klass, object_id_tag, rt_proxy.args
+            )
+            self._realtime_proxies_last_update[rt_proxy.id] = rt_proxy.last_update()
+        except Exception:
+            self.logger.exception('impossible to initialize realtime proxy')
```
Comment (pbench) [inline source/jormungandr/jormungandr/realtime_schedule/realtime_proxy_manager.py]:
```suggestion
            self.logger.exception('impossible to update realtime proxy {}'.format(rt_proxy.id))
```

File: source/jormungandr/jormungandr/realtime_schedule/realtime_proxy_manager.py [hunk 3]
```diff
@@ -31,57 +31,136 @@
...
+
+        # If the realtime proxy added in db is also defined in legacy, delete it.
```
Comment (pbench) [inline source/jormungandr/jormungandr/realtime_schedule/realtime_proxy_manager.py]:
```suggestion
        # If a legacy realtime proxy exists in db, remove it from self._realtime_proxies_legacy
```

File: source/jormungandr/jormungandr/realtime_schedule/realtime_proxy_manager.py (hunk 2 above)
Comment (azime) [reply source/jormungandr/jormungandr/realtime_schedule/realtime_proxy_manager.py]: OK

File: source/jormungandr/jormungandr/realtime_schedule/realtime_proxy_manager.py (hunk 3 above)
Comment (azime) [reply source/jormungandr/jormungandr/realtime_schedule/realtime_proxy_manager.py]: ok

**Review by pbench — APPROVED** [review]

**Pull request merged**